Github support: TEXT mode and UTF-8 support for all objects

gilbertbw opened the pull request:
As discussed in #25, add support to work with a plain folder, rather than SVN. This allows the use of any SCC system (including git). This is refered to as TEXT mode (in contrast to SVN mode).

In addition there is now support for encoding all files as UTF-8, rather than just UCS2 encoded files. This helps when working with SCC tools that expect UTF-8 (rather than CP1252).

These changes make it possible to the the add-in to develop projects hosted on Github.

[AccessSCC.zip](https://github.com/x-ware-ltd/access-scc-addin/files/5509422/AccessSCC.zip)

OliverVahey commented:
Newer version of addin
[AccessSCC.zip](https://github.com/x-ware-ltd/access-scc-addin/files/5511731/AccessSCC.zip)

bkp7 commented on Forms/frmSplash.ACF:
```diff
@@ -112,1121 +112,26 @@ Begin Form
                 0xe12fa329ea7bc540ae94cf4ebf07048a
             End
             Begin
-                Begin Image
```
Please can you put an image of some sort back in. It looks awful without it. Perhaps https://favpng.com/png_view/fork-branching-source-code-version-control-fork-png/mce0WrLp

bkp7 commented on Forms/Access SVN - Options.ACF:
```diff
@@ -1088,1121 +1132,10 @@ Begin Form
                     WebImagePaddingRight =1
                     WebImagePaddingBottom =1
                 End
-                Begin Label
```
Please put an image back - this? https://favpng.com/png_view/fork-branching-source-code-version-control-fork-png/mce0WrLp

bkp7 reviewed (changes requested):
USysRegInfo Table needs to be updated.
Also after creating the new mda you need to populate the 'Company' and 'Comments' fields in the DB Properties section (File/'View and edit database properties'). The values for this need to be decided

gilbertbw commented on README.md:
```diff
@@ -11,7 +11,7 @@ Unlike the Visual Sourcesafe Addin
 *Stop the press*: As of Access 2013, the SCC layer of Access has been "deprecated". This means if you are already using systems that rely on the MSSCCI layer, such as Visual Sourcesafe, [PushOK SCC plugin](http://www.pushok.com/software/svn.html) or [Unified SCC](http://aigenta.com/products/UnifiedScc.aspx), this will no longer work from Access 2013 onwards.
 
-### Recent Release History
+# Recent Release History
```
replaced with link to releases page

gilbertbw commented on README.md:
```diff
@@ -24,10 +24,10 @@ Unlike the Visual Sourcesafe Addin
 - Update to handle regional formatting of numbers. Prior to this release some regions may have caused errors, and would have stored numeric table data incorrectly in the repository. Thanks to Florian Reitmeit for spotting this one.
 
-### Version Control Systems
+# Version Control Systems
```
Done

gilbertbw commented on README.md:
```diff
@@ -24,10 +24,10 @@ Unlike the Visual Sourcesafe Addin
 The only VCS that is supported at the moment is Subversion but future plans could allow support for Git, Mercurial, possibly TFS (Team Foundation Server) and many others if there is enough demand.
 
-### Access Versions
+# Access Versions
```
I think this is correct. We still build a 2002-2003 format database.

bkp7 replied:
A bit more info on later version support plus Ribbons etc. are supported...

gilbertbw replied:
I have updated this in cc701f6906f494f80c43259b514eaa3680b3a180

gilbertbw commented on Modules/modSCC.ACM:
```diff
@@ -5,6 +5,7 @@ Public Enum SccSystem
   sccSystemSubversion = 1  ' Supported
   sccSystemGIT = 2         ' Unsupported
   sccSystemMercurial = 3   ' Unsupported
+  sccSystemText = 4        ' Supported
```
Done

gilbertbw reviewed (commented):
Changes made in response to review

bkp7 commented on README.md:
```diff
@@ -41,20 +41,54 @@ Known issues are:
 - 64 bit Access has never been tested at all (doubt I really want to!)
 
-### Requirements
+# Requirements
```
A bit more info.....

bkp7 commented on Modules/modSCC.ACM:
```diff
@@ -3,9 +3,12 @@ Option Explicit
+  sccSystemText = 4        ' Supported
+  
+  ' These are placeholders for future possible enhanced support
+  ' for specific SCC systems
```
```suggestion
  ' for specific SCC systems. The Text implementation does work generically
  ' with GIT but does not integrate fully, ie. specific git functions (commit, push, etc)
  ' have to be performed by the user
```

gilbertbw replied:
I have added a comment based off of this suggestion

gilbertbw commented:
@bkp7 I think the only outstanding things on this are:

- Decide on a logo
- Decede what to do about the Company and Comments fields

gilbertbw commented:
> 
> 
> @bkp7 I think the only outstanding things on this are:
> 
>     * Decide on a logo
> 
>     * Decede what to do about the Company and Comments fields

I have:
- moved setting Company and Comments fields out into #62 
- added my suggested image. For comparison here is:

1. master
2. as it was before readding the icon
3. now

![image](https://user-images.githubusercontent.com/2099292/110611576-3eda6400-8187-11eb-9dba-017af7b7f29e.png)

here is the addin with these changes:
[AccessSCC.0.69.zip](https://github.com/x-ware-ltd/access-scc-addin/files/6115518/AccessSCC.0.69.zip)

SWMahy commented:
Logo as it is now (3) looks ok to me.

SWMahy reviewed (approved):
Have not been able to perform a proper code review, but I understand I have been using this version of access-scc-addin for development work - with no issues, so have approved on that basis. I think the proposed logo look ok too.
